src/Http.ts

```diff
@@ -57,7 +58,17 @@ export interface HttpRequest<A extends keyof HttpOptions = never> {
   >
 }
 
-export type HttpMethod = 'delete' | 'get' | 'patch' | 'post' | 'put'
+const HttpMethods = [
+  'delete',
+  'get',
+  'head',
+  'options',
+  'patch',
+  'post',
+  'put',
+] as const
+export type HttpMethod = typeof HttpMethods[number]
+export const HttpMethodC = $T.literalUnion(HttpMethods, 'HttpMethod')
```

Comment:
È meglio, quando possibile, derivare il tipo dal codec, così da mantenere una singola sorgente di verità:
```suggestion
export const HttpMethodC = $T.literalUnion(HttpMethods, 'HttpMethod')
export type HttpMethod = t.TypeOf<typeof HttpMethodC>
```